core/flyout_button.js

```diff
@@ -267,7 +276,7 @@ Blockly.FlyoutButton.prototype.onMouseUp_ = function(e) {
       this.targetWorkspace_.getButtonCallback(this.callbackKey_))) {
     console.warn('Buttons should have callbacks. Button text: ' + this.text_);
   } else if (!this.isLabel_) {
-    this.targetWorkspace_.getButtonCallback(this.callbackKey_)(this);
+    this.targetWorkspace_.getButtonCallback(this.callbackKey_)(this, this.callbackArg_);
```

Comment:
I would also update the `Blockly.WorkspaceSvg.prototype.getButtonCallback` jsdoc return function to: 
```js

/**
 * Get the callback function associated with a given key, for clicks on buttons
 * and labels in the flyout.
 * @param {string} key The name to use to look up the function.
 * @return {?function(!Blockly.FlyoutButton, string=)} The function
 *     corresponding to the given key for this workspace; null if no callback is
 *     registered.
 */
```